pandas/core/categorical.py

```diff
@@ -250,20 +250,15 @@ class Categorical(PandasObject):
 
     def __init__(self, values, categories=None, ordered=False, fastpath=False):
 
-        self._validate_ordered(ordered)
-
         if fastpath:
-            # fast path
+            self._dtype = CategoricalDtype(categories, ordered)
             self._codes = coerce_indexer_dtype(values, categories)
-            self._categories = self._validate_categories(
-                categories, fastpath=isinstance(categories, ABCIndexClass))
-            self._ordered = ordered
             return
 
         # sanitize input
         if is_categorical_dtype(values):
 
-            # we are either a Series or a CategoricalIndex
+            # we are either a Series, CategoricalIndex or CategoricalDtype
```

Comment:
this can't be true, values must be a value type and not a dtype (could be a Categorical though I think)